File: 0724-find-pivot-index/0724-find-pivot-index.cpp
class Solution {
public:
    int pivotIndex(vector<int>& nums) {
        int n = nums.size();
        int l = 0;
        int r = 0;
        int pivot = -1;
        int total = 0;

        for(int i = 0; i < n; i++){
            total += nums[i];
        }

        for(int i = 0; i < n; i++){
            r = total - l - nums[i];
            if(l == r){
                pivot = i;
                break;
            }
            l += nums[i];
        }
        return pivot;
    }
};

File: 0724-find-pivot-index/0724-find-pivot-index.py
class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        l = 0
        r = 0
        total = 0
        pivot = -1
        n = len(nums)

        for i in nums:
            total += i

        for i in range(n):
            r = total - l - nums[i]

            if l == r:
                pivot = i
                break

            else:
                l += nums[i]

        return pivot